src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -14310,6 +14874,385 @@ public abstract class MulticastDelegate : Delegate { }
+        }
+
+        [Theory, WorkItem(65728, "https://github.com/dotnet/roslyn/issues/65728")]
+        [InlineData("decimal ")]
+        [InlineData("decimal?")]
+        public void MissingDecimalConstantAttribute_Lambda_CustomType(string type)
+        {
+            var source = $$"""
+                Del lam = ({{type}} d = 1.1m) => { };
+
+                delegate void Del({{type}} d = 1.1m);
+                """;
+            var comp = CreateCompilation(source);
+            comp.MakeTypeMissing(WellKnownType.System_Runtime_CompilerServices_DecimalConstantAttribute);
+            comp.VerifyDiagnostics(
+                // (1,25): error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.DecimalConstantAttribute..ctor'
+                // Del lam = (decimal  d = 1.1m) => { };
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "1.1m").WithArguments("System.Runtime.CompilerServices.DecimalConstantAttribute", ".ctor").WithLocation(1, 25),
+                // (3,32): error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.DecimalConstantAttribute..ctor'
+                // delegate void Del(decimal  d = 1.1m);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "1.1m").WithArguments("System.Runtime.CompilerServices.DecimalConstantAttribute", ".ctor").WithLocation(3, 32));
+        }
+
+        [Theory, WorkItem(65728, "https://github.com/dotnet/roslyn/issues/65728")]
+        [InlineData("decimal", "System.Decimal")]
+        [InlineData("decimal?", "System.Nullable`1[System.Decimal]")]
+        public void MissingDecimalConstantAttribute_Lambda_ExplicitAttribute_Alone(string type, string typeFullName)
+        {
+            var source = $$"""
+                using System.Runtime.CompilerServices;
+
+                var lam = ([DecimalConstant(1, 0, 0u, 0u, 11u)] {{type}} d) => { };
+                System.Console.WriteLine(lam.GetType());
+                var lam2 = lam;
+                System.Console.WriteLine(lam2.GetType());
+                """;
+            var comp = CreateCompilation(source);
+            comp.MakeTypeMissing(WellKnownType.System_Runtime_CompilerServices_DecimalConstantAttribute);
+            CompileAndVerify(comp, expectedOutput: $"""
+                System.Action`1[{typeFullName}]
+                System.Action`1[{typeFullName}]
+                """).VerifyDiagnostics();
+        }
+
+        [Theory, WorkItem(65728, "https://github.com/dotnet/roslyn/issues/65728")]
+        [InlineData("decimal ")]
+        [InlineData("decimal?")]
+        public void MissingDecimalConstantAttribute_Lambda_ExplicitAttribute_WithOptional(string type)
+        {
+            var source = $$"""
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+
+                var lam = ([Optional, DecimalConstant(1, 0, 0u, 0u, 11u)] {{type}} d) => { };
```

Comment:
>var [](http://example.com/codeflow?start=16&length=3)

It feels like for each 'var' scenario with a missing type we should have a similar test with concrete delegate type

Reply:
Thanks, added.